emcc.py

```diff
@@ -859,6 +859,8 @@ def setting_sub(s):
                   break
               libs.append((i, l))
               newargs[i] = ''
+            elif (shared.Settings.WASM_OBJECT_FILES or 'WASM_OBJECT_FILES=1' in settings_changes) and shared.Building.is_wasm(arg):
```

Comment:
seems a little awkward here to check both the settings and if it was changed by the user - is this before `settings_changes` is populated, so you're checking both for a user change and for when we change the default?

Maybe worth adding a comment here that we can remove the complexity when the default changes. Or we could just check for a user change here until we change the default?